node.go

```diff
@@ -284,26 +284,30 @@ func (node *Node) writeBytes(w io.Writer) (n int, err error) {
 func (node *Node) set(t *Tree, key []byte, value []byte) (
 	newSelf *Node, updated bool, orphaned []*Node,
 ) {
+	version := t.version + 1
+
 	if node.isLeaf() {
 		switch bytes.Compare(key, node.key) {
 		case -1:
 			return &Node{
 				key:       node.key,
 				height:    1,
 				size:      2,
-				leftNode:  NewNode(key, value),
+				leftNode:  NewNode(key, value, version),
 				rightNode: node,
+				version:   version,
```

Comment:
interesting, so we just save these on disk (it would help with debugging) but they're not otherwise used? (e.g. not hashed)

Reply:
This was always the case actually, the version of inner nodes was saved but not hashed. The difference is where we set the version. Now we set it here, in set(), previously we would set it on SaveVersion().